ContactDetailsApi/V2/Gateways/ContactDetailsDynamoDbGateway.cs

```diff
@@ -101,5 +114,344 @@ public async Task<ContactDetails> CreateContact(ContactDetailsEntity contactDeta
+                    continue;
+
+                Guid tenureId = (Guid) asset.TenureId;
+
+                try
+                {
+                    var id = tenuresByTenureId[tenureId];
+
+                }
+                catch (Exception e)
+                {
+                    _logger.LogDebug($"tenuresByTenureId for id {tenureId} failed with the following message {e.Message}");
+                    continue;
+                }
+
+                var tenure = tenuresByTenureId[tenureId];
+
+                var personContacts = new List<Person>();
+
+                foreach (var householdMember in tenure.HouseholdMembers)
+                {
+                    try
+                    {
+
+                        var tenant = personById[householdMember.Id];
+                    }
+                    catch (Exception e)
+                    {
+                        _logger.LogDebug($"personById for id {householdMember.Id} failed with the following message {e.Message}");
+                        continue;
+                    }
+                    var person = personById[householdMember.Id];
+
+                    try
+                    {
+                        var contactDetailsTargetId = contactDetailsGroupedByTargetId[person.Id];
+                    }
+                    catch (Exception e)
+                    {
+                        _logger.LogDebug($"contactDetailsGroupedByTargetId for id {person.Id} failed with the following message {e.Message}");
+                        continue;
+                    }
+
+                    var personContactDetails = contactDetailsGroupedByTargetId[person.Id]
+                        .Select(x => new PersonContactDetails
+                        {
+                            ContactType = x.ContactInformation.ContactType.ToString(),
+                            SubType = x.ContactInformation.SubType.ToString(),
+                            Value = x.ContactInformation.Value.ToString()
+                        })
+                        .ToList();
+
+
+                    var personContact = new Person
+                    {
+                        PersonTenureType = householdMember.PersonTenureType.ToString(),
+                        IsResponsible = householdMember.IsResponsible,
+                        FirstName = person.FirstName,
+                        LastName = person.Surname,
+                        Title = person.Title.ToString(),
+                        PersonContactDetails = personContactDetails
+                    };
+
+                    personContacts.Add(personContact);
+                }
+
+                var contactByUprn = new ContactByUprn
+                {
+                    Uprn = asset.Uprn,
+                    TenureId = tenure.Id,
+                    Contacts = personContacts
+                };
+
+                contacts.Add(contactByUprn);
+            }
+            return contacts;
+        }
+
```

Comment:
There are still a lot of try catches that dont do anything.  We should at least log the error. Otherwise we may have missing data that we are not aware of.

Reply:
i've already added a log for all the catches :)

Reply:
I dont see anything on lines 140, 175, 233, 274, 327

Reply:
i've removed those - see my new commits